src/open_inwoner/accounts/views/cases.py

```diff
@@ -153,6 +155,10 @@ def get_context_data(self, **kwargs):
         case = fetch_single_case(case_uuid)
 
         if case:
+            # check if we have a role in this case
+            if not fetch_roles_for_case_and_bsn(case.url, self.request.user.bsn):
+                raise PermissionDenied()
```

Comment:
TBH checking permissions in `get_context_data` seems odd and scattering permission checks between `test_func` and  `get_context_data` makes it hart to understand the authorization logic. 
I'd suggest putting all permissions in one place (`test_func` or `get` or `dispatch`)

Since we already use mixins for permissions and this code is also used in `CaseDocumentDownloadView` it can also be put in the separate mixin